src/components/InPageNavigation/InPageNavigation.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useState } from 'react'
+import classnames from 'classnames'
+import { HeadingLevel } from '../../types/headingLevel'
+import { Link } from '../Link/Link'
+import styles from './InPageNavigation.module.scss'
+
+export type ContentType = [heading: HeadingLevel, text: string, href: string]
+
+type InPageNavigationProps = {
+  className?: string
+  content: JSX.Element
+  headingLevel?: HeadingLevel
+  rootMargin?: string
+  scrollOffset?: string
+  threshold?: number
+  title?: string
+}
+
+export const InPageNavigation = ({
+  className,
+  content,
+  headingLevel,
+  rootMargin,
+  scrollOffset,
+  threshold,
+  title,
+}: InPageNavigationProps): React.ReactElement => {
+  const classes = classnames('usa-in-page-nav', styles.target, className)
+  const Heading = headingLevel || 'h4'
+  const ROOT_MARGIN = '0px 0px 0px 0px'
+  const offsetStyle = {
+    '--margin-offset': scrollOffset || '0',
+  } as React.CSSProperties
```

Comment:
USWDS took a [js approach](https://github.com/uswds/uswds/blob/1b88ad075a7cd3f4eab8a5828244c44995931fe9/packages/usa-in-page-navigation/src/index.js#L141), but this can be done more simply with css. 

Unfortunately, storybook doesn't seem to allow anchor links to work within its nice little frame, so I added another story that shows an egregious 2rem offset:
![2rem](https://github.com/trussworks/react-uswds/assets/764090/3d7afa6d-b445-48e0-a7ec-1daba9a33b6d)

Compared to the default (0):
![zero](https://github.com/trussworks/react-uswds/assets/764090/4a95d000-34ce-45eb-8dcc-4d80a0a44f4c)